db.go

```diff
@@ -690,6 +690,61 @@ type Result interface {
 	// using All().
 	All(sliceOfStructs interface{}) error
 
+	// Paginate splits the results of the query into pages containing pageSize
+	// items.  When using pagination previous settings for Limit and Offset are
+	// ignored.
+	//
+	// Use Page() to define the specific page to get results from.
+	//
+	// Example:
+	//
+	// r = q.Paginate(12)
+	Paginate(pageSize uint) Result
+
+	// Page makes the result set return results only from the page identified by
+	// pageNumber. Page numbering starts from 0.
+	//
+	// Example:
+	//
+	// r = q.Paginate(12).Page(4)
+	Page(pageNumber uint) Result
```

Comment:
Agreed, starting with 1 is more natural for final users and I expect developers to feed this number directly to the pagination API, let's do this.